common/src/main/java/com/gentics/mesh/context/impl/BulkActionContextImpl.java

```diff
@@ -50,18 +50,25 @@ public void process() {
 		process(false);
 	}
 
+	@Override
+	public void setBatch(EventQueueBatch batch) {
+		batch.addAll(this.batch);
+		this.batch = batch;
+	}
+
 	@Override
 	public void process(boolean force) {
 		if (elementCounter.incrementAndGet() >= DEFAULT_BATCH_SIZE || force) {
-			log.info("Processing transaction batch {" + batchCounter.get() + "}. I counted {" + elementCounter.get() + "} elements.");
+			log.info("Processing transaction batch {" + batchCounter.get() + "}. I counted {" + elementCounter.get() + "} elements. {" + asyncActions.size() + "} to be executed.");
 			// Check before commit to ensure we are 100% safe
 			db.blockingTopologyLockCheck();
-			Tx.get().commit();
+			Tx.maybeGet().ifPresent(Tx::commit);
```

Comment:
BAC is allowed to exist outside of the Tx. Every Tx-dependent action is normally executed manually on demand.